test/bin_extract_test.go

```diff
@@ -0,0 +1,271 @@
+package test
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+	"testing"
+
+	"github.com/Azure/dalec"
+	"github.com/Azure/dalec/frontend/pkg/bkfs"
+	gwclient "github.com/moby/buildkit/frontend/gateway/client"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBinExtract_Mariner2(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	testBinExtract(ctx, t, "mariner2/rpm/bin")
+}
+
+func TestBinExtract_AzLinux3(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	testBinExtract(ctx, t, "azlinux3/rpm/bin")
+}
+
+type expectFile struct {
+	contents    string
+	permissions os.FileMode
+}
+
+func assertZipContentsMatch(t *testing.T, zf fs.File, want map[string]expectFile) {
+	info, err := zf.Stat()
+	if err != nil {
+		t.Fatalf("unable to stat zip: %s", err)
+	}
+
+	read, err := io.ReadAll(zf)
+	if err != nil {
+		t.Fatalf("unable to read zip: %s", err)
+	}
+
+	zr, err := zip.NewReader(bytes.NewReader(read), info.Size())
+	if err != nil {
+		t.Fatal("unable to create zip reader %w", err)
+	}
+
+	for _, file := range zr.File {
+		f, err := file.Open()
+		assert.Nil(t, err)
+		contents, err := io.ReadAll(f)
+
+		assert.Nil(t, err)
```

Comment:
```suggestion
		assert.NilError(t, err)
```

Reply:
I don't think that method exists with the testify library

Reply:
Ah that explains it, we have 2 assertion libraries in the repo.

Reply:
Oh interesting. Perhaps we should pare it down to one?